src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/PunishmentsMappingResource.kt

```diff
@@ -57,12 +59,52 @@ class PunishmentsMappingResource(private val mappingService: AdjudicationMapping
     createMappingRequest: AdjudicationPunishmentBatchMappingDto,
   ) =
     try {
-      mappingService.createPunishmentMappings(createMappingRequest)
+      mappingService.createPunishmentMappings(createMappingRequest.punishments)
     } catch (e: DuplicateKeyException) {
       throw DuplicateMappingException(
         messageIn = "Adjudication punishment mapping already exists, detected by $e",
         duplicate = createMappingRequest,
         cause = e,
       )
     }
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_ADJUDICATIONS')")
+  @PutMapping("/mapping/punishments")
```

Comment:
all a bit weird the Rest here - since we must do a batch we have no resource ID so it is the same as the POST but a PUT. No ideal